### PR TITLE
Reorganizing Makefiles

### PR DESCRIPTION
**Pull request by dpetrisko**
This PR targets #107 

Major changes are
1) Get rid of setup_env.sh and replace with Makefile.common at top level
2) Put Makefile.dc, Makefile.vcs, Makefile.verilator in bp_common/syn and referenced from other Ends
(Only BE is done for now)
3) Redirecting logs to bp_\<end\>/syn/logs/\<tool\>/\<tb\>_\<operation\>.log

Still needs to be done
1) Unify all End makefiles
2) change regress scripts to be makefile targets (need to think about how this interacts with CI)
3) Comment all of the makefiles
4) add a default make help target

### CODE REVIEW

**Comment by dpetrisko** [comment]
There are also some synthesis fixes embedded in here, which I ran into when running check_synth with the new flow

File: bp_top/syn/Makefile [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## Set common environment variables
+TOP ?= $(shell git rev-parse --show-toplevel)
+
+include $(TOP)/Makefile.env
```
Comment (stdavids) [inline bp_top/syn/Makefile]: Makefile.common?

**Comment by stdavids** [comment]
Woah, cool git command to get the top level directory, very clever! That will likely be brought to our other infrastructure code :)

One line looked like a bug to me, other than that it looks good to me!

File: bp_top/syn/Makefile (hunk 1 above)
Comment (dpetrisko) [reply bp_top/syn/Makefile]: good catch!

**Comment by dpetrisko** [comment]
@muwyse Could you take a look at this branch? I redid the ME makefiles a bit and I was able to build your bp_me_top_mesi TB, but it was erroring out when I ran it.

"Error: (E109) complete binding failed: port not bound: port 'DUT.port_12' (sc_out)
In file: communication/sc_port.cpp:235"

`make run.sc TB=bp_me_top_mesi` is the command


AFAICT, I kept all of the parameters and verilator flags the same between the builds.  I left your old Makefile and Makefile.verilator as Makefile.old and Makefile.verilator.old for reference

**Comment by muwyse** [comment]
@dpetrisko This error is due to not having updated the ME systemc test bench. It is trying to bind a signal to a port that no longer exists. You can ignore this error for the moment and merge, then I will fixup.

**Comment by dpetrisko** [comment]
Merged into dev instead of be_dev

**Pull request closed without merging**